vidarr-core/src/main/java/ca/on/oicr/gsi/vidarr/core/PriorityConsumableResource.java

```diff
@@ -0,0 +1,140 @@
+package ca.on.oicr.gsi.vidarr.core;
+
+import ca.on.oicr.gsi.Pair;
+import ca.on.oicr.gsi.vidarr.BasicType;
+import ca.on.oicr.gsi.vidarr.ConsumableResource;
+import ca.on.oicr.gsi.vidarr.ConsumableResourceResponse;
+import ca.on.oicr.gsi.vidarr.PriorityFormula;
+import ca.on.oicr.gsi.vidarr.PriorityInput;
+import ca.on.oicr.gsi.vidarr.PriorityScorer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.NullNode;
+import io.undertow.Handlers;
+import io.undertow.server.HttpHandler;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.stream.Collectors;
+
+public final class PriorityConsumableResource implements ConsumableResource {
+  private Integer defaultPriority;
+  private PriorityFormula formula;
+  private Map<String, PriorityInput> inputs;
+  private String name;
+  private PriorityScorer scorer;
+
+  @Override
+  public Optional<HttpHandler> httpHandler() {
+    final var routes = scorer.httpHandler().map(Handlers::path).orElseGet(Handlers::path);
+    for (final var entry : inputs.entrySet()) {
+      entry.getValue().httpHandler().ifPresent(h -> routes.addPrefixPath("/" + entry.getKey(), h));
+    }
+    return Optional.of(routes);
+  }
+
+  @Override
+  public Optional<Pair<String, BasicType>> inputFromSubmitter() {
+    final var type =
+        BasicType.object(
+            inputs.entrySet().stream()
+                .map(e -> new Pair<>(e.getKey(), e.getValue().inputFromSubmitter())));
+    return Optional.of(new Pair<>(name, defaultPriority == null ? type : type.asOptional()));
+  }
+
+  @Override
+  public boolean isInputFromSubmitterRequired() {
+    return defaultPriority == null;
+  }
+
+  @Override
+  public int priority() {
```

Comment:
This `priority` is different than the job `priority`s that the ConsumableResource manages, right?

Reply:
Right.